The initial model included 14 predictors and interaction terms. Although it achieved an R² of 0.803, VIF analysis revealed substantial multicollinearity, particularly among points, shooting attempts, and interaction variables.

In [ ]:
import pandas as pd
import statsmodels.api as sm
# data loading
df = pd.read_csv('//content/nba_team_stats_00_to_23.csv')

#clean columns
df.columns = df.columns.str.strip().str.lower()


# 3. Filter seasons (2013–2023)
# season looks like "2012-13", so extract starting year
df['season_start'] = df['season'].str[:4].astype(int)
df = df[df['season_start'] >= 2013].copy()

# 4. Target variable (win %)

df['win_pct'] = df['wins'] / df['games_played']


# 5. Interaction terms
df['three_interaction'] = df['three_point_percentage'] * df['three_pointers_attempted']
df['fg_interaction'] = df['field_goal_percentage'] * df['field_goals_attempted']


# 6. Features (clean + non-redundant)

features = [
    'points',
    'assists',
    'rebounds',
    'steals',
    'blocks',
    'turnovers',
    'three_point_percentage',
    'three_pointers_attempted',
    'field_goal_percentage',
    'field_goals_attempted',
    'free_throw_percentage',
    'free_throw_attempted',
    'three_interaction',
    'fg_interaction'
]

#prepare data
df_model = df[features + ['win_pct']].dropna()

X = df_model[features]
y = df_model['win_pct']

# Add intercept
X = sm.add_constant(X)


# 8. Fit model

model = sm.OLS(y, X).fit()


# 9. Results

print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                win_pct   R-squared:                       0.803
Model:                            OLS   Adj. R-squared:                  0.795
Method:                 Least Squares   F-statistic:                     91.89
Date:                Tue, 21 Apr 2026   Prob (F-statistic):          3.89e-102
Time:                        03:01:30   Log-Likelihood:                 431.66
No. Observations:                 330   AIC:                            -833.3
Df Residuals:                     315   BIC:                            -776.3
Df Model:                          14                                         
Covariance Type:            nonrobust                                         
                               coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------------
const                   

In [ ]:

# 10. Correlation matrix

corr_matrix = df_model[features].corr()

# Print full matrix
print(corr_matrix)

# Optional: focus on strong correlations
high_corr = corr_matrix[(corr_matrix.abs() > 0.8) & (corr_matrix.abs() < 1.0)]
print("\nHigh correlations (> 0.8):\n", high_corr)

                            points   assists  rebounds    steals    blocks  \
points                    1.000000  0.763179  0.613674  0.240269  0.324497   
assists                   0.763179  1.000000  0.379992  0.256054  0.327267   
rebounds                  0.613674  0.379992  1.000000  0.208869  0.380418   
steals                    0.240269  0.256054  0.208869  1.000000  0.268533   
blocks                    0.324497  0.327267  0.380418  0.268533  1.000000   
turnovers                 0.164625  0.186653  0.419578  0.384059  0.207959   
three_point_percentage    0.297223  0.323177 -0.061433 -0.094296  0.120312   
three_pointers_attempted  0.599255  0.461834  0.129591 -0.097675  0.024301   
field_goal_percentage     0.570213  0.604071  0.004514  0.034168  0.209737   
field_goals_attempted     0.807981  0.604377  0.795860  0.347800  0.314515   
free_throw_percentage     0.243087  0.195183 -0.129384 -0.181220 -0.007845   
free_throw_attempted      0.368061  0.014751  0.478513  0.298598

In [ ]:

# 11. VIF calculation

from statsmodels.stats.outliers_influence import variance_inflation_factor

# Need a DataFrame without the target
X_vif = sm.add_constant(df_model[features])

vif_data = pd.DataFrame()
vif_data["feature"] = X_vif.columns
vif_data["VIF"] = [variance_inflation_factor(X_vif.values, i)
                   for i in range(X_vif.shape[1])]

print(vif_data)

                     feature            VIF
0                      const  302988.850136
1                     points    6387.051642
2                    assists       3.992344
3                   rebounds       4.387086
4                     steals       1.603813
5                     blocks       1.355304
6                  turnovers       1.722910
7     three_point_percentage      27.595552
8   three_pointers_attempted     531.688277
9      field_goal_percentage     399.217912
10     field_goals_attempted    1004.081369
11     free_throw_percentage      54.740746
12      free_throw_attempted     402.017371
13         three_interaction    1138.633385
14            fg_interaction    5731.665502


In [ ]:
# new model


To reduce redundancy, the interaction terms and points were removed. This maintained the model's R² at 0.803 while increasing adjusted R² from 0.795 to 0.796 and substantially reducing VIF values.

In [ ]:
import pandas as pd
import statsmodels.api as sm


# 1. Load data

df = pd.read_csv('//content/nba_team_stats_00_to_23.csv')

# 2. Clean column names

df.columns = df.columns.str.strip().str.lower()

# 3. Filter seasons (2013–2023)

df['season_start'] = df['season'].str[:4].astype(int)
df = df[df['season_start'] >= 2013].copy()

# 4. Target variable (win %)

df['win_pct'] = df['wins'] / df['games_played']


# 5. Interaction terms
#no interactions
# -----------------------------
# 6. Features (clean + non-redundant)

features = [
    'assists',
    'rebounds',
    'steals',
    'blocks',
    'turnovers',
    'three_point_percentage',
    'three_pointers_attempted',
    'field_goal_percentage',
    'field_goals_attempted',
    'free_throw_percentage',
    'free_throw_attempted'
]


# 7. Prepare data

df_model = df[features + ['win_pct']].dropna()

X = df_model[features]
y = df_model['win_pct']

# Add intercept
X = sm.add_constant(X)


# 8. Fit model

model = sm.OLS(y, X).fit()


# 9. Results

print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                win_pct   R-squared:                       0.803
Model:                            OLS   Adj. R-squared:                  0.796
Method:                 Least Squares   F-statistic:                     117.5
Date:                Tue, 21 Apr 2026   Prob (F-statistic):          5.82e-105
Time:                        03:11:24   Log-Likelihood:                 431.04
No. Observations:                 330   AIC:                            -838.1
Df Residuals:                     318   BIC:                            -792.5
Df Model:                          11                                         
Covariance Type:            nonrobust                                         
                               coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------------
const                   

In [ ]:
#VIF OF NEW MODEL MODEL 1.1

In [ ]:
from statsmodels.stats.outliers_influence import variance_inflation_factor

# Need a DataFrame without the target
X_vif = sm.add_constant(df_model[features])

vif_data = pd.DataFrame()
vif_data["feature"] = X_vif.columns
vif_data["VIF"] = [variance_inflation_factor(X_vif.values, i)
                   for i in range(X_vif.shape[1])]

print(vif_data)

                     feature          VIF
0                      const  2734.509971
1                    assists     3.825451
2                   rebounds     4.250672
3                     steals     1.559670
4                     blocks     1.320315
5                  turnovers     1.714419
6     three_point_percentage     1.910698
7   three_pointers_attempted     1.513150
8      field_goal_percentage     2.773549
9      field_goals_attempted     5.395616
10     free_throw_percentage     1.356060
11      free_throw_attempted     1.669638


In [ ]:
#model 1.2 removing blocks

Next, blocks was removed because its coefficient was not statistically significant (p = 0.715). This resulted in only a minimal change in R², from 0.803 to 0.802, while adjusted R² remained at 0.796.

In [ ]:
import pandas as pd
import statsmodels.api as sm

#load data
df = pd.read_csv('//content/nba_team_stats_00_to_23.csv')


# 2. Clean column names

df.columns = df.columns.str.strip().str.lower()

# 3. Filter seasons (2013–2023)

# season looks like "2012-13", so extract starting year
df['season_start'] = df['season'].str[:4].astype(int)
df = df[df['season_start'] >= 2013].copy()

# 4. Target variable (win %)

df['win_pct'] = df['wins'] / df['games_played']


features = [
    'assists',
    'rebounds',
    'steals',
    'turnovers',
    'three_point_percentage',
    'three_pointers_attempted',
    'field_goal_percentage',
    'field_goals_attempted',
    'free_throw_percentage',
    'free_throw_attempted'
]


df_model = df[features + ['win_pct']].dropna()

X = df_model[features]
y = df_model['win_pct']

# Add intercept
X = sm.add_constant(X)


model = sm.OLS(y, X).fit()


print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                win_pct   R-squared:                       0.802
Model:                            OLS   Adj. R-squared:                  0.796
Method:                 Least Squares   F-statistic:                     129.6
Date:                Tue, 21 Apr 2026   Prob (F-statistic):          5.32e-106
Time:                        03:21:07   Log-Likelihood:                 430.97
No. Observations:                 330   AIC:                            -839.9
Df Residuals:                     319   BIC:                            -798.2
Df Model:                          10                                         
Covariance Type:            nonrobust                                         
                               coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------------
const                   

In [ ]:
# model 1.3 removing assists, and free throw attempted

Finally, assists and free-throw attempts were removed because neither was statistically significant in the previous model (p = 0.619 and p = 0.130, respectively). The final model achieved an R² of 0.801 and adjusted R² of 0.796 with 8 predictors.

In [ ]:
import pandas as pd
import statsmodels.api as sm


# 1. Load data

df = pd.read_csv('//content/nba_team_stats_00_to_23.csv')


# 2. Clean column names

df.columns = df.columns.str.strip().str.lower()


# 3. Filter seasons (2013–2023)

# season looks like "2012-13", so extract starting year
df['season_start'] = df['season'].str[:4].astype(int)
df = df[df['season_start'] >= 2013].copy()


# 4. Target variable (win %)

df['win_pct'] = df['wins'] / df['games_played']


#no interactions terms

features = [
    'rebounds',
    'steals',
    'turnovers',
    'three_point_percentage',
    'three_pointers_attempted',
    'field_goal_percentage',
    'field_goals_attempted',
    'free_throw_percentage',

]


# 7. Prepare data

df_model = df[features + ['win_pct']].dropna()

X = df_model[features]
y = df_model['win_pct']

# Add intercept
X = sm.add_constant(X)

#
# 8. Fit model
model = sm.OLS(y, X).fit()

# 9. Results
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                win_pct   R-squared:                       0.801
Model:                            OLS   Adj. R-squared:                  0.796
Method:                 Least Squares   F-statistic:                     161.6
Date:                Tue, 21 Apr 2026   Prob (F-statistic):          1.03e-107
Time:                        03:22:07   Log-Likelihood:                 429.79
No. Observations:                 330   AIC:                            -841.6
Df Residuals:                     321   BIC:                            -807.4
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
                               coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------------
const                   

In [ ]:
# VIF FOR MODEL 1.3 # -----------------------------
from statsmodels.stats.outliers_influence import variance_inflation_factor

# Need a DataFrame without the target
X_vif = sm.add_constant(df_model[features])

vif_data = pd.DataFrame()
vif_data["feature"] = X_vif.columns
vif_data["VIF"] = [variance_inflation_factor(X_vif.values, i)
                   for i in range(X_vif.shape[1])]

print(vif_data)

                    feature          VIF
0                     const  1731.822942
1                  rebounds     3.525109
2                    steals     1.449104
3                 turnovers     1.523003
4    three_point_percentage     1.848706
5  three_pointers_attempted     1.415686
6     field_goal_percentage     1.754870
7     field_goals_attempted     4.164838
8     free_throw_percentage     1.353348


Overall, the model refinement process prioritized a simpler and less redundant model while preserving nearly the same explanatory power.